model/project.go

```diff
@@ -1173,7 +1179,10 @@ func (m *Module) GetRepoOwnerAndName() (string, string) {
 }
 
 // FindTaskGroup returns a specific task group from a project
-func (p *Project) FindTaskGroup(name string) *TaskGroup {
+func (p *Project) FindTaskGroup(name string, group *TaskGroup) *TaskGroup {
+	if group != nil {
```

Comment:
Why have this be part of it? It seems like we just shouldn't call FindTaskGroup if group != nil

Reply:
It seems like FindTaskGroup should do the extra work of finding the task group in the variant list if it's not in the task group list, rather than doing it outside of this.